cmake/onnxruntime_flatbuffers.cmake

```diff
@@ -0,0 +1,19 @@
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License.
+
+file(GLOB onnxruntime_flatbuffers_srcs CONFIGURE_DEPENDS
+    "${ONNXRUNTIME_ROOT}/core/flatbuffers/*.h"
+    "${ONNXRUNTIME_ROOT}/core/flatbuffers/*.cc"
+    )
+
+source_group(TREE ${REPO_ROOT} FILES ${onnxruntime_flatbuffers_srcs})
+
+add_library(onnxruntime_flatbuffers ${onnxruntime_flatbuffers_srcs})
+install(DIRECTORY ${PROJECT_SOURCE_DIR}/../include/onnxruntime/core/flatbuffers  DESTINATION ${CMAKE_INSTALL_INCLUDEDIR}/onnxruntime/core)
+onnxruntime_add_include_to_target(onnxruntime_flatbuffers onnx flatbuffers)
+if(onnxruntime_ENABLE_INSTRUMENT)
+  target_compile_definitions(onnxruntime_flatbuffers PUBLIC ONNXRUNTIME_ENABLE_INSTRUMENT)
+endif()
+target_include_directories(onnxruntime_flatbuffers PRIVATE ${ONNXRUNTIME_ROOT})
+add_dependencies(onnxruntime_flatbuffers ${onnxruntime_EXTERNAL_DEPENDENCIES})
+set_target_properties(onnxruntime_flatbuffers PROPERTIES FOLDER "ONNXRuntime")
```

Comment:
this has a dependency on the graph library doesn't it given flatbuffers_utils.h includes a header from there?

Reply:
may not need a link dependency but would be good to be more explicit about the dependency on the include path

---
In reply to: [494752823](https://github.com/microsoft/onnxruntime/pull/5276#discussion_r494752823) [](ancestors = 494752823)

Reply:
Including  core/graph/basic_types.h is only for the declaration of ONNX_NAMESPACE::ValueInfoProto, removed and add some forward decls



---
In reply to: [494759244](https://github.com/microsoft/onnxruntime/pull/5276#discussion_r494759244) [](ancestors = 494759244,494752823)